enlighten/Controller.py

```diff
@@ -687,6 +688,16 @@ def initialize_new_device(self, device):
         self.multispec.update_hide_others()
         self.multispec.update_color()
 
+        if device.is_andor:
+            andor_eeprom = self.cloud_manager.get_andor_eeprom(spec.settings.eeprom.serial_number.replace("CCD-",""))
+            if andor_eeprom is {}:
+                log.error(f"got empty dict for andor eeprom. Serial number incorrect or no entry in dynamo table.")
+            else:
+                spec.settings.eeprom.excitation_nm_float = andor_eeprom["excitation_nm_float"]
+                spec.settings.eeprom.wavelength_coeffs = andor_eeprom["wavelength_coeffs"]
+                spec.settings.eeprom.detector = spec.setting.eeprom.serial_number
```

Comment:
To clarify, the change requested then is to add a new eeprom field in the ENLIGHTEN GUI?

Reply:
I think we should add a display-only field to the ENLIGHTEN Hardware Setup page, similar to how we currently display values like the firmware revision and current battery charge.  We _should not_ use an EEPROM field to store detector serial number, because we don't need to — the only products for which we need to track detector serial number are the Andor XL models, and we can read the detector serial number for those from the Andor SDK.

My main point is that we should not be writing detector serial number to the eeprom.detector field.  That field should contain the detector partnumber, not detector serial number.